### ISSUE TITLE
Openshot Video Editor/VLC ugly and barely usable

### ISSUE THREAD

**Issue by peterdp** [linuxmint/Cinnamon#8206]
```
 * Cinnamon version (cinnamon --version)

![system_info](https://user-images.githubusercontent.com/46158705/50442334-b2715d00-08fe-11e9-98ca-4167013d42ad.png)

**Issue**

Openshot: blurred icons, very large display of Track editor zone.... 

![ugly_openshot](https://user-images.githubusercontent.com/46158705/50442414-1d229880-08ff-11e9-97ae-f39dc809d02a.png)

VLC: very large display of icons, sub-menu's...:

![ugly_vlc](https://user-images.githubusercontent.com/46158705/50442615-ed27c500-08ff-11e9-9a34-2433fd643fea.png)

**Steps to reproduce**

Issue is always there. Just start up VLC or Openshot on Linux Mint 19.1

**Expected behaviour**

Much smaller, much sharper icons, menu's,....

**Other information**

Possibly a very important general display problem with specific type of UI applications (Qt?)

**Comment by brownsr** [comment]
Sorry @peterdp you need to give us much more information.  When I fire up VLC it looks fine.  So please could you attach screenshots (preferably annotated to make it clear exactly what is catching your eye) of the particular functions you see problems with, give us details like the screen resolution you are using etc.etc.  We have to be able to reproduce your problem accurately first.

**Comment by leigh123linux** [comment]
Post

```
env |grep QT
```

**Comment by peterdp** [comment]
@brownsr I attached screenshots via drag&drop but I only see a ![xxx] (yyyy) kind of text appearing in the issue report. Should I attach screenshots in another way?

**Comment by peterdp** [comment]
@leigh123linux 
env|grep QT
QT_QPA_PLATFORMTHEME=qt5ct
QT_ACCESSIBILITY=1

**Comment by peterdp** [comment]
@brownsr I can get access to the posted screenshots by using the in the issue report mentioned https url directly in a separate tab. Can you try that?

**Comment by peterdp** [comment]
@brownsr 
Used screen resolution:
![screen_resolution](https://user-images.githubusercontent.com/46158705/50443915-bf457f00-0905-11e9-8ea3-94437303ea83.png)

**Comment by peterdp** [comment]
@brownsr If I understand you correctly you've a good looking VLC on Linux Mint Cinnamon 19.1. This is a good thing. This gives us comparison possibilities. Can you post your output of env|grep QT?

**Comment by peterdp** [comment]
@brownsr  Another screenshot to illustrate the icon/menu problem within VLC:

![ugly_vlc_menu](https://user-images.githubusercontent.com/46158705/50445531-b194f700-090f-11e9-85b4-8476ab4ca2af.png)

**Comment by peterdp** [comment]
@brownsr Some annotations of how the issue is observed in Openshot. Red arrows indicating some of the blurred icon/text zones, red box indicating the much too large visualization of the 'Track edit area':  

![ugly_openshot](https://user-images.githubusercontent.com/46158705/50446116-814f5780-0913-11e9-8649-998cd2d20fc9.png)

**Comment by peterdp** [comment]
Also asked help within the Openshot community: https://github.com/OpenShot/openshot-qt/issues/2478

**Comment by Odyseus** [comment]
Hello, everybody.

@peterdp: Assuming that you have qt5ct installed (looking at your environment variables); open qt5ct (Qt5 Settings), choose **gtk2** and apply. The default is **bb10bright** and that screws up the theming of all Qt5 applications.

By the way, did you opened all those Qt applications before installing qt5ct? All my Qt applications looked right (they all use the correct theme set on theme settings) without the need for qt5ct.

**Comment by peterdp** [comment]
@Odyseus I did opened VLC and Openshot directly after a fresh install of Linux Mint 19.1. The problem was there from the beginning. I then searched on the internet for a workaround and installed qt5ct.

In qt5ct the style was already gtk2 and the problem is still there:

![vlc_gtk2](https://user-images.githubusercontent.com/46158705/50556301-aa873380-0cd7-11e9-9037-376fe5c33949.png)

![openshot_gtk2](https://user-images.githubusercontent.com/46158705/50556305-b1ae4180-0cd7-11e9-88be-9b456aa4af82.png)

If I change the style to something else than gtk2 only VLC changes and not Openshot.
The change of the style in VLC however doesn't impact the problem (large menu and icons at bottom are still there).

Any idea which settings I should change to get a similar setup as on your system?

**Comment by smurphos** [comment]
Does messing with the environment variables as outlined here help?

https://forums.linuxmint.com/viewtopic.php?f=47&t=276278

**Comment by NikoKrause** [comment]
Yeah, since Mint 19 the patch for QT themeing has been removed and been replaced with qt5ct.
I like the following solution better:
https://www.linuxuprising.com/2018/05/get-qt5-apps-to-use-native-gtk-style-in.html

**Comment by leigh123linux** [comment]
@peterdp Having to override with qt env's proves the issue is in qt5.

**Comment by ferdnyc** [comment]
At its core, I believe the issue is with something misreporting screen dimensions, and Qt's autoscaling making valiant but ultimately foolhardy attempts to adjust the _physical_ sizes of rendered widgets. We actually hit this issue once before in OpenShot/openshot-qt#1863 — the user ultimately "solved" it by installing a new GPU, and I never stumbled across `QT_AUTO_SCREEN_SCALE_FACTOR`, but it's clearly the same issue.

Observe @peterdp 's gnome-control-center screenshot:
![image](https://user-images.githubusercontent.com/538020/50581765-b5061200-0e2a-11e9-9a01-d45efdd8d38a.png)

That label is the key: **Samsung Electric Company 7"**  I guarantee that, just like in OpenShot issue 1863, the _physical_ size of that screen was being reported as `160x90 mm` with some insanely-high (and false) several-hundred-dpi pixel density. Qt, being trusting and naive, is believing those lies and trying to size the interface so that elements aren't rendered unusably small on the "tiny" screen.

I don't know what's lying about the screen dimensions or why, nor how it could be fixed to report **correct** screen dimensions, but correcting the reported screen size that Qt is working off of would cause its autoscaling to stop making a mess of everything.

**Comment by peterdp** [comment]
@ferdnyc I never used a Wacom tablet on that system. The problem was there after a fresh install of Linux Mint 19.1 without ever having another device connected to the linux system.

To the Linux Mint system only one screen is connected. The connected screen is a Samsung SyncMaster XL2370 (https://www.samsung.com/us/support/owners/product/led-monitor-xl2370-1-series). It's a 23" monitor. 
(to be complete: the XL2370 screen is connected via HDMI to the Linux Mint system. However there's also a VGA cable connected to the screen which I use sometimes to connect a windows portable. Mostly the Linux system is not running when I use the windows portable. However there could have been a moment where both systems were active. The portable is a 14" Dell Latitude E7470. I don't expect this to be important for the problem but one never knows...)  

Output of xrandr --listmonitors:

Monitors: 1
 0: +*HDMI-0 1920/160x1080/90+0+0  HDMI-0
    
Output of qtdiag:

Screens: 1, High DPI scaling: inactive
  0 "HDMI-0" Depth: 24 Primary: yes
  Manufacturer: 
  Model: 
  Serial number: 
  Geometry: 1920x1080+0+0 Available: 1920x1040+0+0
  Physical size: 160x90 mm  Refresh: 60 Hz Power state: 0
  Physical DPI: 304.8,304.8 Logical DPI: 96,96 Subpixel_None
  DevicePixelRatio: 1 Pixel density: 3
  Primary orientation: 2 Orientation: 2 Native orientation: 0 OrientationUpdateMask: 0

LibGL Vendor: NVIDIA Corporation
Renderer: GeForce GTS 450/PCIe/SSE2
Version: 4.6.0 NVIDIA 390.77

So as you already mentioned above the reported physical size of the screen isn't correct (and probably causes all other symptoms). Should be interesting to look into the source code of qtdiag/xrandr to learn how these programs get the physical size. If the source of the wrong info is known one can search further which program is responsible for setting that wrong information. But this of course is easier said than done ;)

**Comment by ferdnyc** [comment]
> Geometry: 1920x1080+0+0 Available: 1920x1040+0+0
> **Physical size: 160x90 mm** Refresh: 60 Hz Power state: 0
> **Physical DPI: 304.8,304.8** Logical DPI: 96,96 Subpixel_None
> DevicePixelRatio: 1 **Pixel density: 3**

Yup, there it is, exactly the same as  OpenShot/openshot-qt#1863 — and even more frustratingly, you're experiencing this with an NVidia GPU, whereas the user there was seeing this problem with a Radeon card, and _replaced_ that with an NVidia to make it go away! So my hopes that this issue might at least be vendor/driver-specific are also crushed. Argh.

**Comment by Nemobil** [comment]
Hi,

I have had the same issue yesterday after a new Mint v19.1 installation on new hardware. The solution that I found was pretty easy. After using a **DVI cable instead of an HDMI cable**, my monitor dimensions have been recognized correctly (24" instead of the faulty 7") and the problem was solved.

I hope this information is helpful for others as well! I would have probably not been able to solve the issue without the help of the good analysis of the issue in this thread! Thanks!

**Comment by flugalgourmet** [comment]
Thought I'd leave my two cents worth. I also have the Samsung monitor that displays a 7" screen even though it is 32". I came from KDE to Cinnamon knowing the problem was in Cinnamon. AFAIK,  the only desktop that does not have the problem is KDE which uses QT so I have a hard time believing that QT is the culprit. By the way, Scribus, VLC, Zeal, OBS Studio, Open Shot Video Editor, and LMMS are affected. Hope this helps someone.

**Comment by ferdnyc** [comment]
> AFAIK, the only desktop that does not have the problem is KDE which uses QT so I have a hard time believing that QT is the culprit.

Oh, Qt definitely isn't the "culprit" so much as it's the _victim_ in all this — Qt applications are the only ones that adjust to physical screen size _by default_ (unless `QT_AUTO_SCREEN_SCALE_FACTOR` is disabled), so they're the only ones affected by the misreported screen dimensions. (All of the applications you listed are Qt applications, I _believe_ — I know for certain VLC, OBS, and OpenShot are.) Gtk applications will obey whatever scaling factor is set in the `*-control-center` preferences, but to my knowledge they don't care about physical display parameters. Which is why most people don't even notice their screen size is being detected incorrectly, until the first time they run a Qt application on that screen.

If the issue doesn't occur when using the KDE desktop environment, it's likely that some GNOME software is responsible for misdetecting the physical dimensions of the display. (I'd be really curious to see the output of `qtdiag` in a KDE session, on a system where the display size is being misreported by GNOME.)

**Comment by flugalgourmet** [comment]
Here is my qtdiag output. My screen size is 32 inches but reports 7.

[qtdiag.txt](https://github.com/linuxmint/cinnamon/files/3335005/qtdiag.txt)

**Comment by ferdnyc** [comment]
@flugalgourmet Even the mentioned Qt applications (VLC, OpenShot, etc...) are properly sized under KDE, despite the `qtdiag` output you posted?

Well, then I'm even more stymied. They must be doing _something_ differently. The only meaningful difference between your `qtdiag` and one from an affected user under Cinnamon (or GNOME Shell) is this section:
```
Theme:
  Platforms requested : unknown,generic
            available : kde,gtk3,qt5ct,lxqt,snap,flatpak,xdgdesktopportal
```
It would seem likely the scaling issue is somewhere in the platform theme modules — not in `kde` (unaffected), but at least `gtk3` and possibly others. Thanks for supplying that helpful data.

**Comment by flugalgourmet** [comment]
Yes. Every app I listed in my original post sizes correctly in KDE even with QT_AUTO_SCREEN_SCALE_FACTOR on. This one thing was keeping me from using other window managers because every one, GNOME, LXDE, LXQT, Cinnamon, MATE, and XFCE4 exhibited the enlarged windows rendering them unuseable. Is it possible that KDE just simply ignores the QT_AUTO_SCREEN_SCALE_FACTOR parameter altogether?

**Comment by ferdnyc** [comment]
> Is it possible that KDE just simply ignores the QT_AUTO_SCREEN_SCALE_FACTOR parameter altogether?

@flugalgourmet Possible, certainly... either ignoring it entirely, or overriding it with a different method of handling DPI-awareness. (There are also other variables that can come into play. Like `QT_SCREEN_SCALE_FACTORS`, which allows the scaling to be adjusted for specific displays.)

These variables are documented in [this Qt doc on High DPI Displays](https://doc.qt.io/qt-5/highdpi.html), where Linux is never once even _mentioned_. Qt's support for HiDPI on Linux platforms is literally undocumented, so it all seems very up-in-the-air unfortunately.

**Comment by flugalgourmet** [comment]
DIFF between qtdiag.kde.root.txt and qtdiag.cinnamon.root.txt.

122c119
<   Geometry: 1360x768+0+0 Available: 1314x738+46+0
---
>   Geometry: 1360x768+0+0 Available: 1305x744+0+0
158a156
> 


DIFF between qtdiag.kde.user.txt and qtdiag.cinnamon.user.txt.

82c82
<   Platforms requested : kde,generic
---
>   Platforms requested : gtk3,gnome,generic
126c126
<   Geometry: 1360x768+0+0 Available: 1360x736+0+0
---
>   Geometry: 1360x768+0+0 Available: 1305x744+0+0

**Comment by flugalgourmet** [comment]
QT_SCREEN_SCALE_FACTORS [list] specifies scale factors for each screen. This will not change the size of point sized fonts. This environment variable is mainly useful for debugging, or to work around monitors with wrong EDID information.
The format can be either a semicolon-separated list of scale factors in the same order as QGuiApplication::screens(), or a semicolon-separated list of name=value pairs, where name is the same as QScreen::name().

@ferdnyc

I wonder if the QT_SCREEN_SCALE_FACTORS list may need to be updated with my monitor and various other Samsungs to remedy this situation?

From the QScreen class documentation:

*manufacturer : const QString
This property holds the manufacturer of the screen
This property was introduced in Qt 5.9.
Access functions:
QString	manufacturer() const


*model : const QString
This property holds the model of the screen
This property was introduced in Qt 5.9.
Access functions:
QString	model() const


*name : const QString
This property holds a user presentable string representing the screen
For example, on X11 these correspond to the XRandr screen names, typically "VGA1", "HDMI1", etc.
Access functions:
QString	name() const


The name sounds interesting since @Nemobil claims that hooking your monitor up with a DVI cable fixes the problem. Still, the fact that only KDE is not affected bothers me. Has anyone checked to see if someone filed a bug report on this. If not, I'll check if nobody wants to.

Also, I seriously doubt that they list every monitor. Perhaps just the ones where the EDID is wrong.

**Comment by ferdnyc** [comment]
> Also, I seriously doubt that they list every monitor.

Oh, they mean every monitor _attached to the system_. As the docs say, `QScreen::name()` is something like "VGA1" or "HDMI1", it's local to the current configuration. `QT_SCREEN_SCALE_FACTORS` can't be shared between systems, or set without knowledge of the currently-attached displays.

**Comment by ferdnyc** [comment]
By pure coincidence, this wandered through my inbox a week or two ago:

[Fedora 31 Self-Contained Change: Qt Wayland By Default On Gnome](https://fedoraproject.org/wiki/Changes/Qt_Wayland_By_Default_On_Gnome)

I'm sure at least some of the reports here were from users running with legacy X sessions, not Wayland. But if KDE sessions _are_ enabling Qt to run in a Wayland-on-Wayland mode by default, as opposed to GNOME-based desktops (where it sounds like Qt always uses an X-based backend _even if_ it's running in a Wayland session), then that could be one of those core differences @flugalgourmet was looking for, the reasons that KDE alone seems to be isolated from this issue.
